code/modules/Economy/AdminTools.dm

```diff
@@ -0,0 +1,109 @@
+/datum/admins/proc/EconomyPanel(action, hrefs)
+    if(!check_rights(R_FUN))	return
+
+    var/dat= "<head><title>Economy Tools</title></head>"
+    var/datum/money_account/detailed_account_view
+    var/creating_new_account
+
+    if(hrefs["econ_panel"] != "open")
+        switch(hrefs["econ_panel"])
+            if("create_account")
+                creating_new_account = 1
+            if("finalise_create_account")
+                var/account_name = hrefs["holder_name"]
+                var/starting_funds = max(text2num(hrefs["starting_funds"]), 0)
+                if ((station_account.money - starting_funds) > 0)
+                    station_account.money -= starting_funds
+                    if(starting_funds >0)
+                        //Create a transaction log entry if you need to
+                        var/datum/transaction/T = new()
+                        T.target_name = account_name
+                        T.purpose = "New account funds initialisation"
+                        T.amount = "([starting_funds])"
+                        T.date = current_date_string
+                        T.time = worldtime2text()
+                        T.source_terminal = "$EE^$%$%ERROR$%#@#"
+                        station_account.transaction_log.Add(T)
+                    create_account(account_name, starting_funds, src)
+                    creating_new_account = 0
+            if("view_account_detail")
+                var/index = text2num(hrefs["account_index"])
+                if(index && index <= all_money_accounts.len)
```

Comment:
accounts db uses random numbers for the index actually so clamping it like this would do nothing